mobile/src/main/java/org/openhab/habdroid/ui/CloudNotificationListFragment.kt

```diff
@@ -147,6 +150,14 @@ class CloudNotificationListFragment : Fragment(), View.OnClickListener, SwipeRef
         }
     }
 
+    // Statusbar notifications should be cleared on visiting notifications page
+    // in the app. Currently clearing all the notifications, but it could be
+    // modified further to remove specific only.
```

Comment:
Doing that is necessary, as we also have other notifications (e.g. for background work). You'd need to add a new method to `CloudMessagingHelper` (e.g. named `clearAllNotifications`) which then probably should call into a static method in `FcmMessageListenerService`, clearing the summary notification + all GCM notifications (see `getGcmNotificationCount` for how to detect this) there.

Reply:
So should I make provision to delete individual notification (by notificationId) as well ?

Reply:
What do you mean exactly? We certainly don't need a way to locally remove individual notifications, but pushing out 'hideNotification' messages for the IDs we clear locally during clear-all seems like a good idea to me. In fact, doing so is already supported by passing `FcmMessageListenerService.SUMMARY_NOTIFICATION_ID` to the notification hiding methods in `FcmRegistrationService`.

Reply:
I've implemented a method in CloudMessagingHelper which clears all the notifications using NotificationManager. I could've reused the code from FirebaseMessagingService for the same, but my implementation adds up only two lines of code. So I suppose this would be fine.

However, one strange thing I noticed is, I'm getting 0 notifications in response to http request made in NotificationListFragment to get all notifications. Is this because I'm using dummy UUID and secret to send notifications from server ? or is this some kind of bug ?

Reply:
How can I get non-null context for calling FcmRegistrationService's static methods ?

Reply:
It depends. If you're in an activitiy, you can use `this` as context. Otherwise you can enter context and Android Studio should suggest a context getter function.
If not, you must pass `context` as argument to the function where you need it.

Reply:
> However, one strange thing I noticed is, I'm getting 0 notifications in response to http request made in NotificationListFragment to get all notifications. Is this because I'm using dummy UUID and secret to send notifications from server ? or is this some kind of bug ?

The notifications send via the web interface are no "real" notifications, thus you don't see them if you enter the Notifications site in the app.

Reply:
If I've to send hideNotificationRequest() using FcmRegistrationService, what would be the senderId ?

Reply:
See https://github.com/openhab/openhab-android/blob/master/mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationService.kt#L107 and its caller.